src/PIL/Image.py

```diff
@@ -980,7 +991,7 @@ def convert_transparency(m, v):
                 else:
                     # get the new transparency color.
                     # use existing conversions
-                    trns_im = Image()._new(core.new(self.mode, (1, 1)))
+                    trns_im = Image._init()._new(core.new(self.mode, (1, 1)))
```

Comment:
`Image._init()._new()` is quite common and very ugly. In this case we don't want to create a new image second time just to copy empty info and palette properties.

I suggest following:

`im._new(core.new())` — creates image like im, copying info and palette. I'm not sure how often this method is used this way. 
`Image._init(core.new())` — creates empty image using only core image object.

The only place I see where we need to create empty Image object (without core object) is `new()` function, but it could be easily avoided.

Reply:
I've cherry-picked #7460 here, since that helps.

I've pushed a commit with a version of your suggestion. It's not exact, because it turns out that `_new()` doesn't just copy existing palettes, it also creates new ones. It's not enough to just attach a palette afterwards in `new()`, because something like `Image.linear_gradient("P")` also needs a palette.